linq4j/src/main/java/org/apache/calcite/linq4j/tree/CastExpression.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.linq4j.tree;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.lang.reflect.Type;
+import java.util.Objects;
+
+/**
+ * Represents an expression cast to the specified type.
+ */
+public class CastExpression extends Expression {
```

Comment:
I thought I left a comment in my previous review that the functionality of Cast already seems to exist: it is a UnaryExpression with type Convert. I don't think we should have two ways to do the same thing unless you can explain clearly how this is different; this will be very hard to maintain, since people won't know where to look for the functionality of casts. Why can't you reuse the Convert?
For some reason I don't see this comment, maybe I forgot to type it.

Reply:
Agree, this was also suggested in https://github.com/apache/calcite/pull/3687#discussion_r1490150823 and adopted in a following commit, so the change was intentional I guess, but if possible let's reuse `convert`

Reply:
Thanks for pointing it out. I was not aware that `Convert` maps onto a cast operation. I removed `CastExpression` and changed the code to use `Expressions.convert_()` instead.